Version 0.4.0 fails to build

The changes in https://github.com/dignifiedquire/async-tar/pull/10 probably need to be reverted.  Attempting to update the library here: https://github.com/swift-nav/esthri/pull/188 results in a compilation error:

```
[2021-07-28T21:21:01.527Z] error[E0433]: failed to resolve: could not find `io` in `async_std`
[2021-07-28T21:21:01.527Z]  --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/async-tar-0.4.0/src/archive.rs:9:5
[2021-07-28T21:21:01.527Z]   |
[2021-07-28T21:21:01.527Z] 9 |     io::prelude::*,
[2021-07-28T21:21:01.527Z]   |     ^^ could not find `io` in `async_std`
[2021-07-28T21:21:01.527Z] 
[2021-07-28T21:21:01.527Z] error[E0432]: unresolved imports `async_std::io`, `async_std::path`, `async_std::prelude`, `async_std::stream`, `async_std::task`
[2021-07-28T21:21:01.527Z]   --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/async-tar-0.4.0/src/archive.rs:8:5
[2021-07-28T21:21:01.527Z]    |
[2021-07-28T21:21:01.527Z] 8  |     io,
[2021-07-28T21:21:01.527Z]    |     ^^ no `io` in the root
[2021-07-28T21:21:01.527Z] 9  |     io::prelude::*,
[2021-07-28T21:21:01.527Z] 10 |     path::Path,
[2021-07-28T21:21:01.527Z]    |     ^^^^ could not find `path` in `async_std`
[2021-07-28T21:21:01.527Z] 11 |     prelude::*,
[2021-07-28T21:21:01.527Z]    |     ^^^^^^^ could not find `prelude` in `async_std`
[2021-07-28T21:21:01.527Z] 12 |     stream::Stream,
[2021-07-28T21:21:01.527Z]    |     ^^^^^^ could not find `stream` in `async_std`
[2021-07-28T21:21:01.527Z] 13 |     task::{Context, Poll},
[2021-07-28T21:21:01.527Z]    |     ^^^^ could not find `task` in `async_std`
```

I attempted to add a build scenario that checks for this error here: https://github.com/silverjam/async-tar/pull/3 -- which shows similar results.  FWIW I tried enabling more features in async_std to get it to compile, but it seems too many things async-tar needs are behind the "default" feature, which also enables the runtime (which is the bit the original PR was trying to avoid).  For example see: https://github.com/async-rs/async-std/blob/master/src/lib.rs#L316 which enables the `fs` and `path` module.

cc @danieleades

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -10,6 +10,7 @@ keywords = ["tar", "tarfile", "encoding"]
 readme = "README.md"
 edition = "2018"
 exclude = ["tests/archives/*"]
+resolver = "2"
 
 description = """
 A Rust implementation of an async TAR file reader and writer. This library does not
@@ -19,7 +20,7 @@ contents are never required to be entirely resident in memory all at once.
 """
 
 [dependencies]
-async-std = { version = "1.6.0", default-features = false }
+async-std = "1.6.0"
 filetime = "0.2.8"
 pin-project = "1.0.8"
 
@@ -36,4 +37,4 @@ xattr = { version = "0.2", optional = true }
 redox_syscall = "0.2"
 
 [features]
-default = ["xattr"]
+default = [ "xattr" ]

